virtualmachine/aws/objectstore.go

```diff
@@ -0,0 +1,109 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/s3"
+)
+
+type S3 struct {
+	Name   string // required
+	Region string // required
+	Prefix string // for creating a bucket /obj1/obj2/obj3
+	// TODO Add support for user details
+}
+
+func (s *S3) GetS3BucketsList() ([]string, error) {
+	// 1. Get s3 service client
+	// 2. Fetch the list
+	var bucketList []string
+
+	region := s.Region
+	if region == "" {
+		region = getRegionFromEnv()
+	}
+
+	s3Svc, err := getS3Client(region)
+	if err != nil {
+		aoerr := fmt.Errorf("Failed to create s3 client")
+		return bucketList, aoerr
+	}
+
+	result, err := s3Svc.ListBuckets(nil)
+	if err != nil {
+		aoerr := fmt.Errorf("Failed in fetching the bucket list: %v", err)
+		return bucketList, aoerr
+	}
+
+	for _, b := range result.Buckets {
+		bucketList = append(bucketList, aws.StringValue(b.Name))
+	}
+
+	return bucketList, nil
+}
+
+func (s *S3) CreateBucket() error {
+
+	region := s.Region
+	if region == "" {
```

Comment:
would it make sense to move this logic of env variable to getS3Client function? -- same goes for GetS3BucketsList() and other functions